db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -2685,7 +2687,11 @@ class NonBatchedOpsStressTest : public StressTest {
     SharedState* const shared = thread->shared;
     assert(shared);
 
-    if (!shared->AllowsOverwrite(key) && shared->Exists(column_family, key)) {
+    const ExpectedValue expected_value =
+        thread->shared->Get(column_family, key);
+    bool may_exist = !ExpectedValueHelper::MustHaveNotExisted(expected_value,
+                                                              expected_value);
+    if (!shared->AllowsOverwrite(key) && may_exist) {
       // Just do read your write checks for keys that allow overwrites.
       return;
```

Comment:
@jowlyzhang - I wonder if we should just check !shared->AllowsOverwrite(key). bool may_exist can later be changed to "true" since this is the read testing and lock is not held.